File: test_planet.py
from planet import Planet, PlanetError
from numbers import Number
import unittest


class PlanetConstructorTest(unittest.TestCase):

    def test_positive_case(self):
        planet = Planet()
        self.assertIsInstance(planet.resolution, Number)

    def test_override_resolution(self):
        resolution = 3.6
        planet = Planet(resolution=resolution)
        self.assertEqual(resolution, planet.resolution)

    def test_invalid_depth(self):
        invalid_resolution = "I am not a numeric value."
        with self.assertRaises(PlanetError):
            Planet(resolution=invalid_resolution)


class GetElevationAtCoordinateTest(unittest.TestCase):

    def test_low_resolution(self):
        planet = Planet(resolution=100000.0)
        elevation = planet.get_elevation_at(lat=-10.0, lon=-43.0)
        self.assertIsInstance(elevation, Number)

    def test_high_resolution(self):
        planet = Planet(resolution=50.0)
        elevation = planet.get_elevation_at(lat=45.0, lon=23.0)
        self.assertIsInstance(elevation, Number)

File: lib/src/test/kotlin/TestPlanet.kt
import com.eliottgray.kotlin.Defaults
import com.eliottgray.kotlin.planet.Planet
import com.eliottgray.kotlin.Point
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class PlanetConstructorTest {

    @Test
    fun test_positive_case(){
        val planet = Planet.get(Defaults.SEED)
        assertEquals(Defaults.SEED, planet.seed)
    }

    @Test
    fun test_custom_params(){
        val seed = 999183.0
        val planet = Planet.get(seed=seed)
        assertEquals(seed, planet.seed)
    }

}

class GetElevationAtCoordinateTest {

    @Test
    fun test_low_resolution(){
        val planet = Planet.get(seed=99987.0)
        val elevation = planet.getElevationAt(lat=-10.0, lon=-43.0, resolution=100000).alt
        assertEquals(-2206.5303409091907, elevation)
    }

    @Test
    fun test_high_resolution(){
        val planet = Planet.get(seed=54399875.0)
        val elevation = planet.getElevationAt(lat=45.0, lon=23.0, resolution=50).alt
        assertEquals(160.4482471060341, elevation)
    }
}

class GetMultipleElevationsTest {

    private lateinit var planet: Planet

    @BeforeEach
    fun setUp(){
        planet = Planet.get(seed=99987.0)
    }

    @Test
    fun test_multiple_points(){
        val points = arrayListOf(
            Point.fromSpherical(lat=-10.0, lon=-43.0, resolution=100000),
            Point.fromSpherical(lat=45.0, lon=23.0, resolution=100000),
            Point.fromSpherical(lat=45.0, lon=23.0, resolution=100000),
            Point.fromSpherical(lat=0.0, lon=0.0, resolution=9876),
            )
        val results = planet.getMultipleElevations(points)
        for (point in results){
            when (point.lat) {
                -10.0 -> {
                    assertEquals(-2206.5303409091907, point.alt, 1.0E-7)
                }
                45.0 -> {
                    assertEquals(point.lat, 45.0)
                    assertEquals(-1272.5419541701758, point.alt, 1.0E-7)
                }
                else -> {
                    assertEquals(point.lat, 0.0)
                    assertEquals(2175.729900945921, point.alt, 1.0E-7)
                }
            }
        }
    }
}
